packages/wrangler/src/__tests__/generate.test.ts

```diff
@@ -62,30 +53,183 @@ describe("generate", () => {
 					Find out more about how to create and maintain Sites projects at https://developers.cloudflare.com/workers/platform/sites.
 					Have you considered using Cloudflare Pages instead? See https://pages.cloudflare.com/."
 				`);
+		});
+
+		it("auto-increments the worker directory name", async () => {
+			fs.mkdirSync("my-worker");
+
+			expect(fs.existsSync("my-worker-1")).toBe(false);
+
+			await expect(
+				runWrangler("generate my-worker worker-typescript")
+			).resolves.toBeUndefined();
+
+			expect(std.out).toStrictEqual(expect.stringContaining("my-worker-1"));
+
+			expect(readDirectory("my-worker-1")).toMatchObject<Directory>({
```

Comment:
To be honest, I don't think this test needs to check the contents of this directory. It is enough just to check that the new directory (with the correct name) exists.
Otherwise, this test becomes brittle to changes to the template that is used to generate the folder, which is not what this test is all about.

Reply:
While you're right that it makes this test brittle, I wanted to write a test that would actually check the contents of the folder because I wasn't sold that we were actually cloning the correct things. In particular, I wanted to make sure that subdirectories were being cloned correctly.